lib/options_validator.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const isObject = require('./utils').isObject;
+
+let validationLevel = 'warn';
+
+function createOptionsSchema(optionsSchema) {
```

Comment:
nit: the schema is actually being provided, you're really making a validator/validation function here